src/main/java/com/vispana/vespa/state/assemblers/ContentAssembler.java

```diff
@@ -146,18 +151,34 @@ private static List<String> fetchSchemas(String configHost, String clusterName)
   }
 
   private static List<Node> fetchDispatcherData(
-      String configHost, String clusterName, String vespaVersion) {
-    if (vespaVersion.startsWith("7")) {
+      String configHost,
+      String clusterName,
+      String vespaVersion,
+      final ApplicationPackage appPackage) {
+
+    String[] version = vespaVersion.split("\\.");
+
+    // Assume semantic versioning major.minor.patch
+    if (version.length != 3) {
+      throw new RuntimeException("Failed to parse vespa version");
+    }
+
+    int majorVersion = Integer.parseInt(version[0]);
+    int minorVersion = Integer.parseInt(version[1]);
+
+    if (majorVersion == 7) {
       var dispatcherUrl =
           configHost + "/config/v1/vespa.config.search.dispatch/" + clusterName + "/search";
       return requestGet(dispatcherUrl, SearchDispatchSchema.class).getNode();
-    } else {
+    } else if (majorVersion == 8 && minorVersion < 323) {
```

Comment:
Not important for now, but we could probably use this new approach for all Vespa 8 (and maybe even Vespa 7) clusters - could be used to cleanup the code. This can be tested later though.